src/_node/apis.ts

```diff
@@ -1,203 +1,16 @@
-};
-export const getNodeDepth = (tree: TNodeTreeData, uid: TNodeUid): number => {
-  let nodeDepth = 0,
-    node = tree[uid];
-
-  while (node.uid !== RootNodeUid) {
-    node = tree[node.parentUid as TNodeUid];
-    ++nodeDepth;
-  }
-
-  return nodeDepth;
-};
-export const getNodeDepthExternal = (
-  tree: TNodeTreeData,
-  uid: TNodeUid,
-): number => {
-  let nodeDepth = 0,
-    node = tree[uid];
-
-  while (node.uid !== RootNodeUid) {
-    node = tree[node.parentUid as TNodeUid];
-    ++nodeDepth;
-  }
-
-  return nodeDepth;
-};
-
-export const getValidNodeUids = (
-  tree: TNodeTreeData,
-  uids: TNodeUid[],
-  targetUid?: TNodeUid,
-  treeType?: string,
-  referenceData?: TNodeReferenceData,
-): TNodeUid[] => {
-  let validatedUids: { [uid: TNodeUid]: boolean } = {};
-
-  // validate collection
-  uids.map((uid) => {
-    // remove parent uids
-    let parentNode = tree[uid];
-    while (parentNode.uid !== RootNodeUid) {
-      parentNode = tree[parentNode.parentUid as TNodeUid];
-      delete validatedUids[parentNode.uid];
-    }
-
-    // remove nested uids
-    Object.keys(validatedUids).map((validatedUid) => {
-      let validatedNode = tree[validatedUid];
-      while (validatedNode.uid !== RootNodeUid) {
-        validatedNode = tree[validatedNode.parentUid as TNodeUid];
-        if (validatedNode.uid === uid) {
-          delete validatedUids[validatedUid];
-          break;
-        }
-      }
-    });
-
-    // add current uid
-    validatedUids[uid] = true;
-  });
-
-  // validate target
-  if (targetUid) {
-    let targetNode = tree[targetUid];
-    while (targetNode.uid !== RootNodeUid) {
-      delete validatedUids[targetNode.uid];
-      targetNode = tree[targetNode.parentUid as TNodeUid];
-    }
-
-    if (treeType === "html") {
-      const { elements } = referenceData as THtmlReferenceData;
-
-      const targetNode = tree[targetUid];
-      const targetNodeData = targetNode.data as THtmlNodeData;
-      const targetRefData = elements[targetNodeData.name];
-
-      if (targetRefData) {
-        if (targetRefData.Contain === "All") {
-          // do nothing
-        } else if (targetRefData.Contain === "None") {
-          validatedUids = {};
-        } else {
-          const containTagObj: { [uid: TNodeUid]: boolean } = {};
-          const tagList = targetRefData.Contain.replace(/ /g, "").split(",");
-          tagList.map((tag: string) => {
-            const pureTag = tag.slice(1, tag.length - 1);
-            containTagObj[pureTag] = true;
-          });
-
-          Object.keys(validatedUids).map((uid) => {
-            const node = tree[uid];
-            const nodeData = node.data as THtmlNodeData;
-            if (!containTagObj[nodeData.name]) {
-              delete validatedUids[uid];
-            }
-          });
-        }
-      }
-    }
-  }
-
-  return uids.filter((uid) => validatedUids[uid]);
-};
+  f_copy,
+  f_create,
+  f_duplicate,
+  f_move,
+  f_remove,
+  n_copy,
+  n_create,
```

Comment:
Why ise f_copy and n_copy, when we can have a proper name like `file_copy` and `node_copy`

Reply:
@atulbhatt-system32 
I already updated it with a new structure. can you have a recheck please? I am pushing now.